xdsl/dialects/mod_arith.py

```diff
@@ -0,0 +1,40 @@
+from abc import ABC
+
+from xdsl.dialects.arith import signlessIntegerLike
+from xdsl.ir import Attribute, Dialect, SSAValue
+from xdsl.irdl import IRDLOperation, irdl_op_definition, operand_def, result_def
+from xdsl.traits import Pure
+
+
+class ModArithOp(IRDLOperation, ABC):
+    # assemblyFormat = "operands attr-dict `:` functional-type(operands, results)"
+    pass
+
+
+class BinaryOp(ModArithOp):
+    lhs = operand_def(signlessIntegerLike)
+    rhs = operand_def(signlessIntegerLike)
+    output = result_def(signlessIntegerLike)
+    # assemblyFormat ="$lhs $rhs attr-dict `:` type($output)"  # operands directive not in asm format
+    traits = frozenset((Pure(),))
+
+    def __init__(self, lhs: SSAValue, rhs: SSAValue, result_type: Attribute | None):
```

Comment:
In a lot of other dialects the type `SSAValue | Operation` is used so that you can pass in single output operations